### PR TITLE
Build Desktop for x64 and arm64 only

### PR DESCRIPTION
**Pull request by kidroca**
cc @roryabraham 

### Details

- Build desktop for 2 mac architectures: x64 and arm64
- Use different artifact (file) names for each output so it does not mess up hosting
- _Changes don't need to be tested on all platforms because they only change the electron builder config / build script_

### Fixed Issues

$ https://github.com/Expensify/App/issues/8004

### Tests

- [ ] Verify you can still build the Desktop App (`npm run desktop-build-staging`)
- [ ] Verify you can still run the Desktop App locally (`npm run desktop`)

#### Contributor (PR Author) Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I clearly indicated the environment tests should be run in (Staging vs Production)
- [x] I wrote testing steps that cover success & fail scenarios (if applicable)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests & verified they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I added comments when the code was not self explanatory
    - [ ] I put all copy / text shown in the product in all `src/languages/*` files (if applicable)
    - [ ] I followed proper naming convention for platform-specific files (if applicable)
    - [ ] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
    - [x] I followed the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs))
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I corroborated the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If I created a new component I verified that a similar component doesn't exist in the codebase

#### PR Reviewer Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [x] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [x] If a new component is created I verified that a similar component doesn't exist in the codebase

### QA Steps

- [ ] Verify you can download the x64 version from `/NewExpensify.dmg`
- [ ] Verify you can download the arm64 version from `/NewExpensify-arm64.dmg`

#### Desktop

Screenshot 2022-03-15 at 18 32 29

##### Build directory content
Screenshot 2022-03-15 at 18 31 40

### CODE REVIEW

File: scripts/build-desktop.sh [hunk 1]
```diff
@@ -24,4 +24,6 @@ info ""
 title "Building Desktop App Archive Using Electron"
 info ""
 shift 1
-"$LOCAL_PACKAGES/electron-builder" --config config/electronBuilder.config.js "$@"
+"$LOCAL_PACKAGES/electron-builder" --config config/electronBuilder.config.js --x64 "$@"
```
Comment (roryabraham) [inline scripts/build-desktop.sh:27]: Can we refactor this to run both builds in parallel in the background? I did something similar with the `.github/scripts/buildActions.sh` that you could use as a reference if needed.

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: scripts/build-desktop.sh (hunk 1 above)
Comment (roryabraham) [reply scripts/build-desktop.sh:27]: Okay, thanks for looking into it.

**Review by roryabraham — APPROVED** [review]

**Comment by mvtglobally** [comment]
@roryabraham  We validated Desktop - Mac x64 version, it is pass, but unfortunately, we dont have Mac arm64 version to QA. Can maybe someone check internally?

**Comment by johnmlee101** [comment]
Downloaded and it opened just fine 👍

**Comment by Santhosh-Sellavel** [comment]
Working fine for me too!
Screenshot 2022-03-28 at 10 29 55 PM

**Pull request merged**